state_file.py

```diff
@@ -0,0 +1,67 @@
+import os
+
+class StateFile:
+    default_dict = {
+        "pre-step" :"offline",
+        "masters"  :"offline",
+        "workers"  :"offline",
+        "post-step":"offline",
+        "cluster-name":"",
+    }
+
+    def __init__(self):
+        self.f = open("state_file", "a+")
+        self.f.seek(0)
+
+        stats = os.stat("state_file")
+        if not stats.st_size:
+            self.state_dict = self.default_dict
+            self.write_dict()
+        else:
+            self.state_dict = self.to_dict()
+        return
+    def insert_field(self, name: str, value: str) -> None :
+        out = name + ":" + value + '\n'
+
+        self.f.seek(0, os.SEEK_END)
+        self.f.write(out)
+        return
+    def dinsert(self, name: str, value: str) -> None:
+        self.state_dict[name] = value
+        self.write_dict()
+        self.f.flush()
+        return
+    def read(self) -> str:
+        self.f.seek(0)
+        s = self.f.read()
+        return s
+    def to_dict(self):
+        self.f.seek(0)
+        dict = {}
+        while s := self.f.readline():
+            s = s.strip()
+            key, value = s.split(":")
+            dict[key] = value
+        return dict
+    def not_deployed(self, name: str) -> bool:
+        if self.state_dict[name] == "offline": return True
+    def write_dict(self):
+        self.f.truncate(0)
+        for key, value in self.state_dict.items():
+            self.insert_field(key, value)
+        return
+    def clear_state(self) -> None:
+        self.f.truncate(0)
+        self.state_dict = self.default_dict
+        return
+    def default_values(self) -> None:
+        self.f.truncate(0)
+        return
+    def formatted(self) -> None:
+        print("\nCluster Deployment State:")
+        for key, value in self.state_dict.items():
+            print(f"{key:12}: {value:}")
+        return
+    def __del__(self):
+        self.f.close()
```

Comment:
Almost like https://docs.python.org/3/library/json.html ? Can we reduce the amount of code we have to maintain by using a lib?